n2survey/lime/survey.py

```diff
@@ -458,19 +462,89 @@ def plot(
         theme = deep_dict_update(theme, theme_kwargs)
 
         question_type = self.get_question_type(question)
+        # get plot title
+        if plot_title:
+            if type(plot_title) != str:
+                plot_title = self.get_label(question)
 
         if question_type == "single-choice":
             counts_df = self.count(question, labels=True)
 
             if "title" not in non_theme_kwargs:
                 non_theme_kwargs.update({"title": counts_df.columns[0]})
 
-            fig, ax = single_choice_bar_plot(
-                x=counts_df.index.values,
-                y=pd.Series(counts_df.iloc[:, 0], name="Number of Responses"),
-                theme=theme,
-                **non_theme_kwargs,
-            )
+            if compare_with:
+                DF = []
+                if self.get_question_type(compare_with) == "single-choice":
+
+                    # create Dataframe from both questions
+                    DF.append(
+                        pd.concat(
+                            [
+                                self.get_responses(
+                                    question, labels=True, drop_other=True
+                                ),
+                                self.get_responses(
+                                    compare_with, labels=True, drop_other=True
+                                ),
+                            ],
+                            axis=1,
+                        ).values
```

Comment:
It's not really that one wants to keep any part pandas only. It's perfectly fine to use np where needed, but rather that this specific action you want to perform here, with `pd.concat()` instead of `df.append()`, can be done without a detour to an ndarray, and maybe save us a picosecond of compute and also 1 femtogram of CO2 emission :)